diff --git a/dist/runtime/internal/app.mjs b/dist/runtime/internal/app.mjs
index 943b77e08909ae4b520b9cffbb0008486c03ab72..12a59d69a0a69bdae8a8d2ae79f189758cc0031c 100644
--- a/dist/runtime/internal/app.mjs
+++ b/dist/runtime/internal/app.mjs
@@ -25,7 +25,7 @@ function createNitroApp() {
   const config = useRuntimeConfig();
   const hooks = createHooks();
   const captureError = (error, context = {}) => {
-    const promise = hooks.callHookParallel("error", error, context).catch((error_) => {
+    const promise = hooks.callHookParallel("error", error, context)?.catch((error_) => {
       console.error("Error while capturing another error", error_);
     });
     if (context.event && isEvent(context.event)) {
@@ -74,17 +74,17 @@ function createNitroApp() {
       event.captureError = (error, context) => {
         captureError(error, { event, ...context });
       };
-      await nitroApp.hooks.callHook("request", event).catch((error) => {
+      await nitroApp.hooks.callHook("request", event)?.catch((error) => {
         captureError(error, { event, tags: ["request"] });
       });
     },
     onBeforeResponse: async (event, response) => {
-      await nitroApp.hooks.callHook("beforeResponse", event, response).catch((error) => {
+      await nitroApp.hooks.callHook("beforeResponse", event, response)?.catch((error) => {
         captureError(error, { event, tags: ["request", "response"] });
       });
     },
     onAfterResponse: async (event, response) => {
-      await nitroApp.hooks.callHook("afterResponse", event, response).catch((error) => {
+      await nitroApp.hooks.callHook("afterResponse", event, response)?.catch((error) => {
         captureError(error, { event, tags: ["request", "response"] });
       });
     }
